forest/main.py

```diff
@@ -29,10 +29,15 @@
 
 def main(argv=None):
     args = parse_args.parse_args(argv)
+    print(args)
     if len(args.files) > 0:
         config = cfg.from_files(args.files, args.file_type)
     else:
-        config = cfg.load_config(args.config_file)
+        config = cfg.Config.load(
+                args.config_file,
+                variables=cfg.combine_variables(
```

Comment:
I'm not sure why this is being supplied as a keyword argument?

I'd also be tempted to make the `combine_variables` behaviour built in to `Config.load`.

Reply:
Interface segregation, combining `os.environ`, `args.variables`, `query_params`, etc. into a single dict is a separate responsibility to templating. I would be tempted to hoist the i/o out of load to make it easier to test but then we start to lose the convenience and conciseness of `.load`

Reply:
We might not agree on what makes a more sensible segregation, but there's still no reason to supply the variables as a keyworg argument.